### PR TITLE
extend pre-commit

### LINKED ISSUE #4289: move lints to pre-commit
### 🛠 Proposed Refactor

I would suggest moving most of the extra linting from https://github.com/pyg-team/pytorch_geometric/blob/master/.github/workflows/linting.yml
to recommit as you already use the bot and set the check as mandatory :rabbit: 

### Suggest a potential alternative/fix

check that can be safely moved to pre-commit:
- [flake8](https://flake8.pycqa.org/en/latest/user/using-hooks.html)
- [yamllint](https://yamllint.readthedocs.io/en/stable/integration.html)
- sort - it is already there
- [pylint](https://pylint.pycqa.org/en/latest/user_guide/pre-commit-integration.html)
- [pyroma](https://github.com/regebro/pyroma#version-control-integration)

Also, happy to send a PR if you agree with it... :rabbit:

### PR DESCRIPTION
**Pull request by Borda**
Resolves #4289

### CODE REVIEW

File: .github/workflows/linting.yml [hunk 1]
```diff
@@ -8,85 +8,8 @@ on:  # yamllint disable-line rule:truthy
 
 jobs:
 
-  flake8:
```
Comment (rusty1s) [inline .github/workflows/linting.yml:11]: Any reason to remove them here? As fas as I know, installation of `pre-commit` is not required, right?

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -2,6 +2,23 @@ ci:
...
+  - repo: https://github.com/pre-commit/pre-commit-hooks
+    rev: v4.1.0
+    hooks:
+      - id: end-of-file-fixer
+      - id: trailing-whitespace
+      #- id: check-docstring-first
+      - id: check-json
+      - id: check-yaml
+      - id: check-toml
+      #- id: check-case-conflict
```
Comment (rusty1s) [inline .pre-commit-config.yaml:14]: Do these checks (`docstring-first` and `case-conflict`) fail at the moment?

**Review by rusty1s — APPROVED** [review]

File: .pre-commit-config.yaml (hunk 2 above)
Comment (Borda) [reply .pre-commit-config.yaml:19]: no, they just were not part of my suggestion so was feeling hesitant to include them without your approval :rabbit:

File: .github/workflows/linting.yml (hunk 1 above)
Comment (Borda) [reply .github/workflows/linting.yml:11]: they are just duplicates to what the pre-commit does, so you want to jeep both?
note that your pre-cimmot[bot] is called on each PR too

Comment (Borda) [reply .github/workflows/linting.yml:11]: image

Comment (rusty1s) [reply .github/workflows/linting.yml:11]: Yes, thanks for clarifying. It is indeed good to remove them.

**Comment by Borda** [comment]
Btw, have noticed that updating YAPF to the latest v0.32 will generate a significant portion of changes

**Comment by rusty1s** [comment]
Yes, please leave it at 0.31 for now :)

File: .github/workflows/linting.yml (hunk 1 above)
Comment (rusty1s) [reply .github/workflows/linting.yml:11]: The required checks should be resolved.

**Comment by Borda** [comment]
> Yes, please leave it at 0.31 for now :)

yes I will, just for the record, the pre-commit[bot] has set bu default auto-update so it may happen very soon that it create a PR here with updating all hooks to the latest version including YAPF v0.32

**Pull request merged**